### PR TITLE
Optimize reduce_polys_base

### PR DESCRIPTION
**Pull request by 0x0ece**
Make `reduce_polys_base` parallel.

Unfortunately I had to use two collect(), not sure if there's a better way to do this.
In all practical cases I analyzed, collect() wasn't an issue as the number of polynomials is relatively small (256, 500, up to 800ish). But parallelization improves significantly the perf.

### CODE REVIEW

**Comment by 0x0ece** [comment]
Mmm... I noticed that collecting self.base.powers() oom many tests. I don't have an easy solution here, prob best to close this, unless anyone has an idea?

File: plonky2/src/util/reducing.rs [hunk 1]
```diff
@@ -82,15 +84,15 @@ impl<F: Field> ReducingFactor<F> {
 
     pub fn reduce_polys_base<BF: Extendable<D, Extension = F>, const D: usize>(
         &mut self,
-        polys: impl IntoIterator<Item = impl Borrow<PolynomialCoeffs<BF>>>,
+        polys: Vec<&PolynomialCoeffs<BF>>,
```
Comment (dlubarov) [inline plonky2/src/util/reducing.rs:87]: Could this take something like a `impl MaybeParIter`, to avoid collecting?

**Comment by 0x0ece** [comment]
Disregard the comment above, it turns out that I was doing par_iter().zip(x.par_iter()), while instead doing zip().collect().par_iter() seems to work fine.

File: plonky2/src/util/reducing.rs (hunk 1 above)
Comment (0x0ece) [reply plonky2/src/util/reducing.rs:87]: I'm not really able to make it compile without a Vec. Tried a bunch of variations...
Anyway the issue with collect is fixed.

**Comment by Nashtare** [comment]
Hey @0x0ece, apologies for never getting back to you!
I think we would still be interested in merging this, if performances are indeed improved significantly. Could you rebase with latest `main`, so that we can do some performance benchmarks between both branches?

**Pull request closed without merging**